src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackageCommandRunner.cs

```diff
@@ -0,0 +1,450 @@
+                }
+
+                var projectName = project.GetPropertyValue(ProjectName);
+
+                var assetsPath = project.GetPropertyValue(ProjectAssetsFile);
+
+                //If the file was not found, print an error message and continue to next project
+                if (!File.Exists(assetsPath))
+                {
+                    Console.Error.WriteLine(string.Format(CultureInfo.CurrentCulture,
+                        Strings.Error_AssetsFileNotFound,
+                        projectPath));
+                    Console.WriteLine();
+                    continue;
+                }
+
+
+                var lockFileFormat = new LockFileFormat();
+                var assetsFile = lockFileFormat.Read(assetsPath);
+
+                //Get all the packages that are referenced in a project
+                var packages = msBuild.GetResolvedVersions(listPackageArgs.Frameworks, assetsFile, listPackageArgs.IncludeTransitive);
+
+                //No packages means that no package references at all were found 
+                if (!packages.Any())
+                {
+                    Console.WriteLine(listPackageArgs.Frameworks.Count() == 0 ? string.Format(Strings.ListPkg_NoPackagesFound, projectName) : string.Format(Strings.ListPkg_NoPackagesFoundForFrameworks, projectName));
+                }
+                else
+                {
+                    //Handle outdated
+                    if (listPackageArgs.IncludeOutdated)
+                    {
+                        packages = await AddLatestVersions(packages, listPackageArgs);
+                    }
+
+                    //Printing packages of a single project and keeping track if
+                    //an auto-referenced package was printed
+                    var autoRefFoundWithinProject = false;
+                    PrintProjectPackages(packages, projectName, listPackageArgs.IncludeTransitive, listPackageArgs.IncludeOutdated, out autoRefFoundWithinProject);
+                    autoReferenceFound = autoReferenceFound || autoRefFoundWithinProject;
+                }
+
+                //Unload project
+                ProjectCollection.GlobalProjectCollection.UnloadProject(project);
+            }
+
+            //If any auto-references were found, a line is printed
+            //explaining what (A) means
+            if (autoReferenceFound)
+            {
+                Console.WriteLine(Strings.ListPkg_AutoReferenceDescription);
+            }
+        }
+
+        /// <summary>
+        /// Fetches the latest versions for all of the packages that are
+        /// to be listed
+        /// </summary>
+        /// <param name="packages">The packages found in a project</param>
+        /// <param name="listPackageArgs">List args for the token and source provider</param>
+        /// <returns>A data structure like packages, but includes the latest versions</returns>
+        private async Task<IEnumerable<FrameworkPackages>> AddLatestVersions(
```

Comment:
any async function should have Async suffix in it's name. Follow this practice across your code.